### PR TITLE
Updated and added new test-cases for Exception Handling implementation 

### PR DESCRIPTION
**Pull request by shafreenAnfar**

### CODE REVIEW

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java [hunk 1]
```diff
@@ -28,146 +34,174 @@
...
 
         // CarbonMessage
-        CarbonMessage faultyCarbonMessage = Mockito.mock(CarbonMessage.class);
-        Mockito.when(faultyCarbonMessage.isFaulty()).thenReturn(true);
-        Mockito.when(faultyCarbonMessage.getNelException())
-                .thenReturn(Mockito.mock(ConnectionTimeoutException.class));
+        CarbonMessage faultyCarbonMessage = mock(CarbonMessage.class);
+        when(faultyCarbonMessage.isFaulty()).thenReturn(true);
+        when(faultyCarbonMessage.getNelException())
+                .thenReturn(mock(ConnectionTimeoutException.class));
```
Comment (sdkottegoda) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: I think we can have this in a single line.

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java [hunk 2]
```diff
@@ -28,146 +34,174 @@
...
         new FlowControllerExceptionCallback(handleException1, tryBlockMediatorCant, variableStack,
-                Mockito.mock(DefaultExceptionHandler.class)).done(faultyCarbonMessage);
+                mock(DefaultExceptionHandler.class)).done(faultyCarbonMessage);
 
-        Mockito.verify(defaultExceptionHandler, Mockito.times(1)).handleException(
+        verify(defaultExceptionHandler, times(1)).handleException(
                 faultyCarbonMessage, carbonCallback);
     }
 
     @Test
```
Comment (sdkottegoda) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: Shall we describe the test method?

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java (hunk 1 above)
Comment (shafreenAnfar) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: Done. Updated all required code lines.

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java [hunk 3]
```diff
@@ -28,146 +34,174 @@
...
-        DefaultExceptionHandler defaultExceptionHandler = Mockito.mock(DefaultExceptionHandler.class);
+        DefaultExceptionHandler defaultExceptionHandler = mock(DefaultExceptionHandler.class);
 
         // chain with can't handle scenario
         CarbonCallback handleException1 = new FlowControllerExceptionCallback(carbonCallback,
                 tryBlockMediatorCant, variableStack, defaultExceptionHandler);
         new FlowControllerMediateCallback(handleException1, mediator, variableStack).done(faultyCarbonMessage);
 
-        Mockito.verify(defaultExceptionHandler, Mockito.times(1)).handleException(
+        verify(defaultExceptionHandler, times(1)).handleException(
```
Comment (shafreenAnfar) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: Done. Updated all required code lines.

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java [hunk 4]
```diff
@@ -28,146 +34,174 @@
...
+        when(((TryBlockMediator) tryBlockMediatorCant).hasExceptionHandler())
                 .thenReturn(true).thenReturn(false);
-        Mockito.when(((TryBlockMediator) tryBlockMediatorCant).popHandler()).thenReturn(new CustomExceptionHandler());
+        when(((TryBlockMediator) tryBlockMediatorCant).popHandler()).thenReturn(new CustomExceptionHandler());
 
-        DefaultExceptionHandler defaultExceptionHandler = Mockito.mock(DefaultExceptionHandler.class);
+        DefaultExceptionHandler defaultExceptionHandler = mock(DefaultExceptionHandler.class);
 
         // chain with can't handle scenario
         CarbonCallback handleException1 = new FlowControllerExceptionCallback(carbonCallback,
```
Comment (shafreenAnfar) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: Yeah. handleException1 is not needed. Hence changed it to handleException.

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java (hunk 2 above)
Comment (shafreenAnfar) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: Added a description to each method.

**Pull request merged**